token/services/ttxdb/db/badger/badger.go

```diff
@@ -512,6 +521,91 @@ func (db *Persistence) GetStatus(txID string) (driver.TxStatus, error) {
 	return driver.Unknown, errors.Errorf("transaction [%s] not found", txID)
 }
 
+func (db *Persistence) AddTransactionEndorsementAck(txID string, id view.Identity, sigma []byte) error {
+	key, err := db.transactionEndorseAckKey(txID, id)
+	if err != nil {
+		return errors.Wrapf(err, "could not get key for token request %s", txID)
+	}
+	logger.Debugf("adding transaction endorsement ack [%s] with key [%s]", txID, key)
+
+	value := &TransactionEndorseAck{
+		TxID:  txID,
+		ID:    id,
+		Sigma: sigma,
+	}
+
+	b, err := MarshalTransactionEndorseAck(value)
+	if err != nil {
+		return errors.Wrapf(err, "could not marshal record for key %s", key)
+	}
+
+	txn := db.db.NewTransaction(true)
+	defer txn.Discard()
+	err = txn.Set([]byte(key), b)
+	if err != nil {
+		return errors.Wrapf(err, "could not set value for key %s", key)
+	}
+	err = txn.Commit()
+	if err != nil {
+		return errors.Wrapf(err, "could not commit transaction for [%s]", txID)
+	}
+
+	return nil
+}
+
+func (db *Persistence) GetTransactionEndorsementAcks(txID string) (map[string][]byte, error) {
+	acks := make(map[string][]byte)
+
+	// search for all matching keys
+	type Entry struct {
+		key   string
+		value []byte
+	}
+	var entries []Entry
```

Comment:
I see that another method `SetStatus()` and `GetStatus()` use the same logic for the streams. Maybe you could extract this functionality into a method:
```
func (db *Persistence) GetEntriesByTxID(txID string, prefix string) []Entries {
 var entries []Entry
	stream := db.db.NewStream()
	stream.NumGo = db.numGoStream
	stream.LogPrefix = streamLogPrefixStatus
	txIdAsBytes := []byte(txID)
	stream.ChooseKey = func(item *badger.Item) bool {
		return bytes.HasSuffix(item.Key(), txIdAsBytes)
	}
if len(prefix) > 0 { stream.Prefix = []byte(prefix)}
	stream.Send = func(buf *z.Buffer) error {
		list, err := badger.BufferToKVList(buf)
		if err != nil {
			return err
		}
		for _, kv := range list.Kv {
			entries = append(entries, Entry{key: string(kv.Key), value: kv.Value})
		}
		return nil

	}
	if err := stream.Orchestrate(context.Background()); err != nil {
		return err
	}
return entries
}
```